### PR TITLE
pythonPackages.{hiredis,aioredis,channels_redis,django-crispy-forms,django-filter}: init; Added baddecisionsalex to maintainers.

### PR DESCRIPTION
**Pull request by alex-ameen**
###### Motivation for this change

Required packages for a project at work.

###### Things done

* adding BadDecisionsAlex to maintainer's list.
* pythonPackages.django-filter: init at 2.1.0
* pythonPackages.django-crispy-forms: init at 1.7.2
* pythonPackages.channels_redis: init at 2.4.0
* pythonPackages.aioredis: init at 1.2.0
* pythonPackages.hiredis: init at 1.0.0

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by worldofpeace** [comment]
We have an eval failure
```
while evaluating the attribute 'maintainers' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-0-gleber.ewr1.nix.ci/pkgs/development/libraries/rocksdb/default.nix:41:5:
undefined variable 'magenbluten' at /var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/mr-est/eval-0-gleber.ewr1.nix.ci/pkgs/development/libraries/rocksdb/default.nix:41:44
```

Looks like a entry was removed from the maintainers list in https://github.com/NixOS/nixpkgs/pull/63545/commits/59446b97da8a014e6a96466288fcd3209aa3ede1

File: pkgs/development/python-modules/aioredis/default.nix
Comment (risicle) [inline pkgs/development/python-modules/aioredis/default.nix]: I would explicitly set `doCheck = false` with an explanatory comment that it relies on the presence of a redis server.

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (risicle) [inline pkgs/development/python-modules/channels_redis/default.nix]: To enable tests:

```
  src = fetchFromGitHub {
    owner = "django";
    repo = "channels_redis";
    rev = version;
    sha256 = "05niaqjv790mnrvca26kbnvb50fgnk2zh0k4np60cn6ilp4nl0kc";
  };

  checkInputs = [ pytest async_generator ];
  checkPhase = ''
    # ensure tests are running against installed package
    mv channels_redis channels_redis.hidden
    # test_double_receive expects a redis server running
    pytest tests/ -k "not test_double_receive"
  '';
```

File: pkgs/development/python-modules/django-crispy-forms/default.nix
Comment (risicle) [inline pkgs/development/python-modules/django-crispy-forms/default.nix]: This could be said about most of the test suites we use. Doesn't stop us, because enabling tests makes the package collection far more robust.

**Comment by BadDecisionsAlex** [comment]
Hey I was pretty swamped the last few days. I’ll fix the maintainers list. Since it was the first commit is it possible to rebase it back there without needing to redo all of the later ones?
Im by no means a rebase veteran 😂

File: pkgs/development/python-modules/django-filter/default.nix
Comment (risicle) [inline pkgs/development/python-modules/django-filter/default.nix]: `ERROR: django-filter requires Python '>=3.4' but the running Python is 2.7.16`

You'll have to add `disabled = pythonOlder "3.4";` to prevent it trying to build this package.

**Comment by risicle** [comment]
There's no hurry.

Probably the least confusing way of editing underlying commits when they're quite well separated is to add (again, separate) commits on top of the existing ones with temporary commit comments (e.g. `TMP aioredis`) and then doing an interactive rebase (`git rebase -i <base commit>`) to reorder the commits and `fixup` the temporary ones, folding them into their corresponding original commits.

**Comment by BadDecisionsAlex** [comment]
Okay I got the pulled in the maintainer's list from master and modified that old commit. I don't know if "adding" the changes from master is problematic. If it is let me know and I can re-edit the commits.

I am just seeing the rest of these conversations about other fixes. I will work my way through those now.

File: pkgs/development/python-modules/aioredis/default.nix
Comment (BadDecisionsAlex) [reply pkgs/development/python-modules/aioredis/default.nix]: I got that added. Thanks for the tip.

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (BadDecisionsAlex) [reply pkgs/development/python-modules/channels_redis/default.nix]: Is it necessary to pull the GitHub version to get tests working? Or can I just add the `checkInputs` and `checkPhase` portions?

File: pkgs/development/python-modules/django-crispy-forms/default.nix
Comment (BadDecisionsAlex) [reply pkgs/development/python-modules/django-crispy-forms/default.nix]: How should I proceed with resolving this?
The testing procedures that they have written come with instructions for creating a django app before testing. I could create a repo that contains that skeleton and pull it potentially as an additional source.

File: pkgs/development/python-modules/django-filter/default.nix
Comment (BadDecisionsAlex) [reply pkgs/development/python-modules/django-filter/default.nix]: Got that one knocked out.

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (BadDecisionsAlex) [reply pkgs/development/python-modules/channels_redis/default.nix]: Another one bites the dust!

@risicle thank you so much for all of the help!

File: pkgs/development/python-modules/django-crispy-forms/default.nix
Comment (risicle) [reply pkgs/development/python-modules/django-crispy-forms/default.nix]: Yeah it's quite likely it's not feasible to get them going, it was just an offhand remark tbh.

Comment (BadDecisionsAlex) [reply pkgs/development/python-modules/django-crispy-forms/default.nix]: Word.

**Comment by BadDecisionsAlex** [comment]
@FRidh I think that might have wrapped this one up.
Let me know if you want any additional revisions.

**Comment by risicle** [comment]
I think you might need to take another stab at the "add to maintainers" commit, making it only add yourself.

**Comment by risicle** [comment]
(`nox-review` is happy for me now, non-nixos linux x86_64)

**Comment by BadDecisionsAlex** [comment]
> I think you might need to take another stab at the "add to maintainers" commit, making it only add yourself.

Hey I finally got that fixed :)

File: pkgs/development/python-modules/aioredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/aioredis/default.nix]:
```suggestion
    description = "asyncio (PEP 3156) Redis client library";
```

File: pkgs/development/python-modules/django-filter/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/django-filter/default.nix]:
```suggestion
    description = "Reusable Django application for allowing users to filter querysets dynamically";
```

File: pkgs/development/python-modules/hiredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/hiredis/default.nix]: This is too long and shouldn't have a period.

File: pkgs/development/python-modules/django-filter/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/django-filter/default.nix]:
```suggestion
    homepage = "https://github.com/carltongibson/django-filter";
```

File: pkgs/development/python-modules/hiredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/hiredis/default.nix]:
```suggestion
    homepage = "https://github.com/redis/hiredis-py";
```

File: pkgs/development/python-modules/hiredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/hiredis/default.nix]: This is the three clause https://github.com/redis/hiredis-py/blob/master/COPYING

File: pkgs/development/python-modules/hiredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/hiredis/default.nix]:
```suggestion
}:

```

File: pkgs/development/python-modules/hiredis/default.nix
Comment (worldofpeace) [reply pkgs/development/python-modules/hiredis/default.nix]: https://github.com/redis/hiredis-py/blob/master/setup.py#L21

File: pkgs/development/python-modules/django-filter/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/django-filter/default.nix]: three clause https://github.com/carltongibson/django-filter/blob/master/LICENSE

File: pkgs/development/python-modules/django-filter/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/django-filter/default.nix]:
```suggestion
}:

```

File: pkgs/development/python-modules/django-filter/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/django-filter/default.nix]:
```suggestion
  disabled = isPy27;
```

File: pkgs/development/python-modules/django-crispy-forms/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/django-crispy-forms/default.nix]:
```suggestion
    homepage = "https://github.com/maraujop/django-crispy-forms";
```

File: pkgs/development/python-modules/django-crispy-forms/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/django-crispy-forms/default.nix]:
```suggestion
}:

```

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/channels_redis/default.nix]:
```suggestion
    homepage = "https://github.com/django/channels_redis";
```

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/channels_redis/default.nix]:
```suggestion
```

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/channels_redis/default.nix]:
```suggestion
}:

```

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/channels_redis/default.nix]: Please normalize `pname` to `channel-redis`.

File: pkgs/top-level/python-packages.nix
Comment (worldofpeace) [inline pkgs/top-level/python-packages.nix]: Please rename this attribute to `channel-redis` and the directory.

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/channels_redis/default.nix]: three clause https://github.com/django/channels_redis/blob/master/LICENSE

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/channels_redis/default.nix]:
```suggestion
    description = "Redis channel layer backend for Django Channels";
```

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/channels_redis/default.nix]: Please use a `disabled` parameter so this package can be disabled for the following python versions https://github.com/django/channels_redis/blob/master/setup.py#L32

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/channels_redis/default.nix]: I'm seeing more testing requirements here https://github.com/django/channels_redis/blob/2.4.0/setup.py#L12 (specifically cryptography).

Are then not needed in the code anymore?

File: pkgs/development/python-modules/aioredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/aioredis/default.nix]:
```suggestion
    homepage = "https://github.com/aio-libs/aioredis";
```

File: pkgs/development/python-modules/aioredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/aioredis/default.nix]:
```suggestion

  disabled = isPy27;

```

File: pkgs/development/python-modules/aioredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/aioredis/default.nix]:
```suggestion
, hiredis, async-timeout, isPy27
```

File: pkgs/development/python-modules/aioredis/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/aioredis/default.nix]: You need to add this conditioned on `!isPyPy` https://github.com/aio-libs/aioredis/blob/ba9f5b707a29df5c90bc0a1c242d835564407894/setup.py#L9

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Left some suggestions that, when implemented, should be acceptable for merge :+1: 

I'd recommend reading the suggestions commit by commit.

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (BadDecisionsAlex) [reply pkgs/development/python-modules/channels_redis/default.nix]: Can I confirm this one?
In this case the package is actually named `channels_redis` (plural with underscore, not dash) consistently in their documentation, homepages, and in dependency lists for all `requirements.txt` files.
Are you sure that we want to alter their convention?

Comment (worldofpeace) [reply pkgs/development/python-modules/channels_redis/default.nix]: I meant `channels-redis`.

And yes we do, there's an ongoing effort to normalize the derivation name in this manner and
the nixpkgs documentation on python packages states that attributes should be normalized similarally.

Comment (worldofpeace) [reply pkgs/development/python-modules/channels_redis/default.nix]: For reference:

See https://nixos.org/nixpkgs/manual/#contributing
> Attribute names in python-packages.nix should be normalized according to PEP 0503. This means that characters should be converted to lowercase and . and _ should be replaced by a single - (foo-bar-baz instead of Foo__Bar.baz ) 

And 
> https://nixos.org/nixpkgs/manual/#sec-package-naming
states that dashes are preferred.

I can't recall where the discussion for python packages is specifically.

File: pkgs/development/python-modules/channels_redis/default.nix
Comment (risicle) [reply pkgs/development/python-modules/channels_redis/default.nix]: The path that uses `cryptography` (https://github.com/django/channels_redis/blob/6f6ced8afa126b533fef32461b2ecfd1f8f5a0a3/channels_redis/core.py#L256) doesn't appear to be covered by the tests.

Comment (worldofpeace) [reply pkgs/development/python-modules/channels_redis/default.nix]: yep it is an extra feature
* https://github.com/django/channels_redis/blob/6f6ced8afa126b533fef32461b2ecfd1f8f5a0a3/setup.py#L39

I'd leave that up to the author to decide whether they want it enabled default or not.

**Comment by FRidh** [comment]
Nothing happening so closing.

**Pull request closed without merging**
